patroni/postgresql.py

```diff
@@ -1645,52 +1717,77 @@ def basebackup(self, conn_url, env):
 
         return ret
 
-    def pick_synchronous_standby(self, cluster):
-        """Finds the best candidate to be the synchronous standby.
-
-        Current synchronous standby is always preferred, unless it has disconnected or does not want to be a
-        synchronous standby any longer.
-
-        :returns tuple of candidate name or None, and bool showing if the member is the active synchronous standby.
+    def current_sync_state(self, cluster):
+        """Returns current synchronous replication state as a dict:
+        * numsync: number of synchronous nodes requested(including leader)
+        * sync: set of nodes potentially being synced to
+        * active: set of nodes that are sync capable
         """
-        current = cluster.sync.sync_standby
-        current = current.lower() if current else current
+        sync_standby_names = self.query("SHOW synchronous_standby_names")
```

Comment:
as I can see here https://github.com/zalando/patroni/blob/master/patroni/postgresql.py#L1509
`self.query` - will return `cursor`, not the result of query execution. And after that You are passing cursor to 
`sync_rep_parser_re.finditer` its not acceptable.